MsCorePkg/Library/PlatformBootManagerLib/BdsPlatform.c

```diff
@@ -388,6 +388,14 @@ PlatformBootManagerAfterConsole (
     case BOOT_ON_FLASH_UPDATE:
       EfiBootManagerConnectAll ();
       DEBUG ((DEBUG_INFO, "[%a] - signalling capsules are ready for processing\n", __FUNCTION__));
+      DEBUG ((DEBUG_INFO, "[%a] - Deleting Memory Type Information variable for capsule update\n", __FUNCTION__));
+      Status = gRT->SetVariable (

```

Comment:
The return from `SetVariable()` assigned to `Status` should be checked.

Within the scope of the BDS code, I think it should be considered that the variable doesn't exist. Trying to delete a variable that does not exist should return `EFI_NOT_FOUND`.

To satisfy both of these, I think `GetVariable()` should first be called to test for existence of the variable. If present, proceed to call `SetVariable()` with zero-sized data to delete the variable. In that case an error is not expected from the `SetVariable()` call so add an `ASSERT_EFI_ERROR (Status)` to catch that in assert-enabled builds.

Reply:
To add on that, the variable attribute should come from the query of `GetVariable` instead of hardcoded. Otherwise, the deletion might fail.

Reply:
Should be updated now.